리뷰 2주차

BeigeBearDev opened the pull request:
멘토링

rOyalFruit committed:
Fix: 도서 상세 페이지 에러 수정

- BookDetailPage 컴포넌트에서 params 비동기 처리 구현
- BookDetailPageProps 인터페이스 업데이트: params를 Promise 타입으로 변경
- fetch 호출 시 await로 해결된 id 값 사용
- Route used `params.id`. `params` should be awaited before using its properties" 에러 해결

Commit:
fix: memberId를 orderId로 변경하여 주문 조회 수정

rOyalFruit committed:
Remove application.yml from git history

rOyalFruit committed:
feat: 장바구니 추가 기능 구현 (회원/비회원)

- 회원: 임시로 쿠키의 member_id를 사용하여 서버에 장바구니 추가 요청
  - user_session, member_id 쿠키 존재 시 로그인 상태로 간주
  - POST /cart/{book-id}/{member-id} 엔드포인트 호출

- 비회원: 로컬스토리지를 사용하여 장바구니 데이터 관리
  - 키: guest_cart
  - 데이터: bookId, quantity 정보 저장
  - 동일 상품 추가 시 수량 증가 처리

추가 구현 사항:
- 장바구니 담기 버튼 컴포넌트 분리
- 수량 선택 및 총 상품금액 표시 기능

Commit:
fix: 프론트 코드에서 데이터 불러오기 수정

Commit:
fix: 주문 상세 페이지 수정

Commit:
chore: 초기 데이터 수정 - memberId=2의 주문 3개 추가

pcyscott committed:
OAuth-back

Commit:
fix: 주문 상태 및 배송 상태 수정

rOyalFruit committed:
Feat: openapi-typescript를 이용해서 통신에 필요한 타입 스크립트 생성

- @Profile("dev")로 dev 모드일 때만 실행됨
- ./frontend/types/schema.d.ts 경로에 생성됨

1m1nkim committed:
Feat: 검색 시 정렬 기준, 별점 기능

- 출간일순, 판매량순, 평점순, 리뷰순 정렬
- 평점순은 총점인 db로는 구현하지 못해서 averageRating 추가
- 정렬 시 판매량인 경우 판매량이 같거나 없는경우 출간일 순으로 보조 정렬, 다른 것도 같음

pcyscott committed:
login1-front

1m1nkim committed:
Merge remote-tracking branch 'origin/Feature1-Book,Cart' into Feature1-Book,Cart

# Conflicts:
#	backend/src/main/resources/application.yml

rOyalFruit committed:
Fix: Merge conflict resolved

1m1nkim committed:
Feat: Bookinfo 별점 추가, CartTest 개발

rOyalFruit committed:
Fix: Merge conflict resolved

1m1nkim committed:
Feat: 검색 시 제목, 저자, 출판사, ISBN13 등 검색 기능 구현
Refactor: BaseInitDate makeSampleBooks() 일부 수정

BeigeBearDev committed:
feat: 전체 주문 조회 기능 추가 및 코드 리팩토링

- 전체 주문 조회 API 추가
- 기존 서비스, 컨트롤러, DTO를 관리자 전용으로 분리
- order 엔티티와 orderDetail 엔티티 간 1:N 관계 적용
- 코드 컨벤션 적용 및 구조 개선
- 페이징 및 정렬 기능 추가

Commit:
Merge branch 'dev' of https://github.com/prgrms-be-devcourse/NBE3-4-2-team8 into feature-order-search

pcyscott committed:
change to JWT -back

pcyscott committed:
simplify auth

pcyscott committed:
modify controller with jwt

nodonghui committed:
fect : qna API 구현

question 컨트롤러와 answer 컨트롤러 api를 구현했습니다.

1m1nkim committed:
Merge remote-tracking branch 'refs/remotes/origin/Feature1-Book,Cart' into Feature1-Book,Cart

# Conflicts:
#	frontend/app/components/book/BookInfo.tsx

pcyscott committed:
jwt at frontend

BeigeBearDev committed:
Merge pull request #17 from prgrms-be-devcourse/feature-admin-14

Refactor: 전체 주문 조회 기능 개선 및 구조 리팩토링

rOyalFruit committed:
Merge pull request #18 from prgrms-be-devcourse/feature-login

Feat: 카카오API 로그인, JWT토큰 활용

BeigeBearDev committed:
Revert "Feat: 카카오API 로그인, JWT토큰 활용"

rOyalFruit committed:
Merge pull request #19 from prgrms-be-devcourse/revert-18-feature-login

Revert "Feat: 카카오API 로그인, JWT토큰 활용"

BeigeBearDev committed:
Merge branch 'dev' into feature-login

BeigeBearDev committed:
Merge pull request #20 from prgrms-be-devcourse/feature-login

카카오 로그인 및 JWT 기반 인증 구현

rOyalFruit committed:
Merge remote-tracking branch 'origin/dev' into Feature1-Book,Cart

# Conflicts:
#	backend/src/main/resources/application.yml
#	frontend/app/components/NavBar.tsx
#	frontend/package.json

rOyalFruit committed:
Fix: 브랜치 병합 후 발생한 에러 수정

rOyalFruit committed:
Fix: 카카오 로그인을 수행해도 쿠키가 구워지지 않는 문제 임시로 수정

BeigeBearDev committed:
feat: 관리자 주문 상세내역 조회 기능 추가 및 리팩토링

- 주문 상세내역 조회 API 추가
- DrtailOrder 엔티티 안 DeliveryStatus ENUM 파일 분리
- 코드 컨벤션 적용 및 구조 개선

BeigeBearDev committed:
feat: 관리자 주문 상세내역 조회 기능 추가 및 리팩토링

- 주문 상세내역 조회 API 추가
- DrtailOrder 엔티티 안 DeliveryStatus ENUM 파일 분리
- 코드 컨벤션 적용 및 구조 개선

rOyalFruit committed:
Refactor: 장바구니 추가 기능 수정

- jwtToken 쿠키로 로그인 여부를 확인하도록 수정
- 백엔드 단에서 쿠키 httpOnly(true) 설정을 임시로 제거.

Commit:
Merge remote-tracking branch 'origin/dev' into feature-order-search

nodonghui committed:
refactor : 코드 리팩토링

코드 구조 변경

nodonghui committed:
작업 내용 저장 (예: QnA 기능 수정)

nodonghui committed:
Conflict 해결

Commit:
fix: 충돌 해결

Commit:
Remove application.yml from repository tracking

Commit:
fix: 서버 요청 및 설정 오류 수정

1. 프론트에서 서버로 요청을 해야되는데 서버가 안 돌고 있었음.
    1. ExternalBookAPIService -> WebClientBuilder Bean을 불러와야 되는데, Bean에 등록되어 있지 않아서 서비스 초기화 시에 에러가 남.
    2. application.yml의 aladin.ttb 키 위치가 잘못되어 있었음. aladin.ttbkey여야 되는데 custom.aladin.ttbkey로 되어 있었음.
2. 프론트에서 서버로 book, event, image 관련 요청을 해야 되는데, SecurityConfig에서 인증된 유저만 호출할 수 있도록 설정되어 있었음.
    1. securityConfig 담당하는 사람한테 열어달라고 요청함.
3. 서버에서 보낸 example.com/cover.image가 존재하지 않아서 렌더링 시에 오류가 나서 프론트 서버가 실행되지 않았음.
    -> 임시로 next.config.ts에 관련 설정 추가.

rOyalFruit committed:
Refactor: pr멘토링 리뷰 사항 반영, 장바구니 기능 리팩토링

- Cart 유저 인증 방식 변경
- cart api를 테스트하려면 jwtToken 쿠키 발급 필요.

Commit:
refactor: PR 리뷰 반영

Commit:
commit

pcyscott committed:
using token-back

pcyscott committed:
adding cookie usage at front

BeigeBearDev committed:
Merge pull request #22 from prgrms-be-devcourse/feature-admin-15

Feat: 관리자 주문 상세내역 조회, 배송상태 수정 기능 추가 및 리팩토링

nodonghui committed:
Merge pull request #25 from prgrms-be-devcourse/feature-token

Feature token

nodonghui committed:
save

Commit:
fix: conflict 해결 및 병합 완료

rOyalFruit committed:
Fix: dev 브랜치 병합 후 생긴 에러 수정 - 1

rOyalFruit committed:
Fix: SpringDoc 濡蕤� �� 臾몄� ��

Commit:
Merge pull request #21 from prgrms-be-devcourse/Feature1-Book,Cart

Feature1 book,cart

Commit:
feat: memberid 필드를 oauthid로 수정

Commit:
fix: 충돌 해결 후 병합

rOyalFruit committed:
Refactor: 사용자 인증 정보를 받아오는 방식 변경

- Cart, Review 컨트롤러에 해당 내용 반영됨

1m1nkim committed:
Refactor: 프론트 인증 방식 변경 MemberDto Id 값 추가
Fix: 프론트에서 SortType 이름 바뀐 것 적용

-  리뷰 등록, 수정, 삭제 memberid 값으로 비교

BeigeBearDev committed:
Merge remote-tracking branch 'origin/dev' into feature-admin-16

# Conflicts:
#	backend/src/main/java/com/ll/nbe342team8/domain/order/detailOrder/entity/DetailOrder.java
#	backend/src/main/java/com/ll/nbe342team8/global/config/SwaggerConfig.java

Commit:
fix: 오류 해결

pcyscott committed:
regenerate token && answer 1

nodonghui committed:
merge finish

nodonghui committed:
merge finsh

1m1nkim committed:
Refactor: 이벤트 배너 첫 실행시에만 저장되게 리팩토링

- PostConstruct 처리 하여 실행 시에만 리스트로 저장 후 리스트로 프론트의 캐시 값으로 비교

nodonghui committed:
로컬 dev 브랜치 업데이트

nodonghui committed:
before merge

rOyalFruit committed:
Feat: 로그인 시 로컬 스토리지(비로그인 상태) 장바구니 내역을 DB에 업데이트

rOyalFruit committed:
Fix: Merge conflict resolved

nodonghui committed:
bug fix

nodonghui committed:
Merge pull request #27 from prgrms-be-devcourse/Feature1-Book,Cart

Feature1 book,cart

nodonghui committed:
merge finish

nodonghui committed:
Merge pull request #29 from prgrms-be-devcourse/feature/review/nodonghui

Feature/review/nodonghui

nodonghui committed:
Revert "Feature/review/nodonghui"

nodonghui committed:
Merge pull request #30 from prgrms-be-devcourse/revert-29-feature/review/nodonghui

Revert "Feature/review/nodonghui"

pcyscott committed:
token managing

pcyscott committed:
delete tokens && redirection

Commit:
fix: detailorder oauthid 수정

BeigeBearDev committed:
feat: Admin Book 관련 기능 추가 및 리팩토링

BeigeBearDev committed:
feat: Admin Login 관련 기능 추가 및 리팩토링

BeigeBearDev committed:
Merge branch 'dev' into feature-admin-16

nodonghui committed:
Merge pull request #32 from prgrms-be-devcourse/feature-admin-16

Feature: 관리자 페이지 도서, 로그인, 구문 관련 기능 추가 및 리팩토링

rOyalFruit committed:
Refactor: Cart 로그인/비로그인 사용자 확인 로직 변경

- Cart 엔티티에 유니크 제약조건 설정
- 로그인 상태를 확인하는 방법:
  - 기존: 서버측 401 응답 여부로 확인
  - 수정: 프론트 로그인 상태를 전역으로 관리

BeigeBearDev committed:
Revert "Feature: 관리자 페이지 도서, 로그인, 구문 관련 기능 추가 및 리팩토링"

BeigeBearDev committed:
Merge pull request #33 from prgrms-be-devcourse/revert-32-feature-admin-16

Revert "Feature: 관리자 페이지 도서, 로그인, 구문 관련 기능 추가 및 리팩토링"

rOyalFruit committed:
Merge pull request #34 from prgrms-be-devcourse/Feature1-Book,Cart

Refactor: Cart 로그인/비로그인 사용자 확인 로직 변경

BeigeBearDev committed:
feat: Admin 관련 기능 추가 및 리팩토링

nodonghui committed:
fect : qna,mypage 수정

Commit:
fix:

nodonghui committed:
save

Commit:
fix: 변경 사항 커밋

nodonghui committed:
refactor : front refactor

nodonghui committed:
fix: bug fix

rOyalFruit committed:
Feat: 결제 기능 컨트롤러 추가

pcyscott committed:
qna back2

nodonghui committed:
Merge pull request #36 from prgrms-be-devcourse/pr/nodonghui

Pr/nodonghui

BeigeBearDev committed:
Merge pull request #35 from prgrms-be-devcourse/feature-admin-fix

feat: Admin 관련 기능 추가 및 리팩토링

1m1nkim committed:
Feat: 주문기능 프론트엔드 생성

Commit:
fix: 병합 후 충돌 해결

pcyscott committed:
resolving merge

pcyscott committed:
refacoring

rOyalFruit committed:
Feat: 주문 화면에 카카오 우편번호 서비스 도입

1m1nkim committed:
Feat: 바로 구매 기능 구현

Commit:
fix: 마이페이지 버튼

1m1nkim committed:
Design: 프론트 레이아웃 적용

1m1nkim committed:
Merge branch 'dev' into Feature1-Book,Cart

pcyscott committed:
Merge pull request #38 from prgrms-be-devcourse/feature-admin-qna

 Admin QnA managing

Commit:
Merge pull request #39 from prgrms-be-devcourse/Feature1-Book,Cart

Feature1 book,cart

Commit:
fix: 병합 후 충돌 해결

Commit:
fix: oauthid 제거 후 member로 수정

rOyalFruit committed:
Fix: 肄� 蹂 �¦ �릿 �믕 ��

rOyalFruit committed:
怨듬� �� �¦�黝��

Commit:
fix:수정

Commit:
.

Account-4-Review commented on backend/build.gradle:
```diff
@@ -25,10 +25,13 @@ repositories {
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'org.springframework.boot:spring-boot-starter-webflux'
     implementation 'org.springframework.boot:spring-boot-starter-validation'
+    implementation("io.jsonwebtoken:jjwt-api:0.11.5")
```
형식은 맞춰주세요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/admin/service/AdminQuestionService.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    //문자열이 비어있는지 확인
+    private boolean isBlank(String str) {
```
저번에 라이브러리 설명드렸었죠?
그리고 그게 아니더라도, 다른곳에서 사용할 가능성이 있다면 이 메서드는 여기 있으면 안됩니다.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/admin/service/AdminQuestionService.java:
```diff
@@ -0,0 +1,72 @@
+            return questionRepository.findByTitleContainingOrContentContaining(keyword, keyword, pageable);
+        }
+        return hasAnswer
+                ? questionRepository.findByAnswersIsNotEmptyAndTitleContainingOrContentContaining(keyword, keyword, pageable)
```
메서드 이름의 가독성이 정말, 정말 안 좋네요...
아무리 Data JPA의 네이밍 컨벤션이라고 해도, 적당히 자를 필요가 있습니다.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/controller/AdminBookController.java:
```diff
@@ -0,0 +1,101 @@
+	public ResponseEntity<String> deleteBook(@PathVariable Long bookId) {
+		adminBookService.deleteBook(bookId);
+
+		return ResponseEntity.ok("도서가 성공적으로 삭제되었습니다.");
```
가능하면, 기업의 OpenAPI 명세를 읽어보면서, 이쪽에서 제공하는 리턴값을 꼭 확인해봤으면 좋겠어요.
- https://developers.kakaopay.com/docs/moneytransfer/sendmoney.code/api-qrlink-delete
- https://developers.worksmobile.com/kr/docs/group-delete

공통점이 있지 않나요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/dto/BookResponseDto.java:
```diff
@@ -1,56 +1,107 @@
+				.author(this.author != null ? this.author : "")                              // 필수: 저자
+				.publisher(this.publisher)                                                   // 선택: 출판사
+				.isbn(this.isbn)                                                             // 선택: ISBN
+				.isbn13(this.isbn13 != null ? this.isbn13 : "9999999999999")                 // 필수: ISBN13
```
꼭 기본 placeholder가 있어야 할까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/entity/Book.java:
```diff
@@ -1,108 +1,104 @@
+	private Integer pricesSales;         // 판매가
 
-    private String toc;        // 목차
+	@NotNull
```
전반적인 indent가 꼬인 것 같아요.

nodonghui committed:
fect : qna 질문 프론트 구현

nodonghui committed:
Conflict resolved

nodonghui committed:
merge

nodonghui committed:
Merge pull request #42 from prgrms-be-devcourse/feature-qna_mypage-nodonghui-2025-02-11

qna 프론트 질문 페이지 완성

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/AdminBookService.java:
```diff
@@ -0,0 +1,123 @@
+	public boolean registerBook(String isbn13) {
+		// 기존 DB에서 중복 확인
+		if (bookRepository.existsByIsbn13(isbn13)) {
+			throw new IllegalArgumentException("ISBN " + isbn13 + "은(는) 이미 등록된 도서입니다.");
```
이거 문자열 합치는 법 다음에 알려드릴게요 ㅎㅎ

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/BookService.java:
```diff
@@ -1,113 +1,93 @@
+		return bookRepository.count();
+	}
+
+	public Book create(Book book) {
```
Transactional 이 없네요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/BookService.java:
```diff
@@ -1,113 +1,93 @@
+		}
+
+		switch (searchType) {
+			case AUTHOR:
```
JDK 17로 넘어가면서 switch 구문이 어떻게 변했는지 알아보시면 좋을 것 같습니다.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/ExternalBookApiService.java:
```diff
@@ -1,38 +1,63 @@
+			JsonNode itemNode = rootNode.path("item").get(0); // 첫 번째 도서 정보 추출
+
+			if (itemNode.isMissingNode()) {
+				System.out.println("API 응답에 도서 정보가 없습니다.");
```
무슨 일이 있어도 서버 개발에선 System.out.println 을 쓰지 말아주세요.

Account-4-Review reviewed (approved):
프로젝트를 보면서 든 생각들을 간단하게 적어봅니다.

- 처음에 비해서 프로젝트의 컨벤션이 어느정도 잡힌 것 같아요. 이 부분은 잘 유지해주셨으면 좋겠습니다.
- API 설계 관점에서 어려움을 겪은 듯한 느낌이 드는데요. 리뷰 코멘트에서 남겼듯이 다른 기업의 openapi 스펙을 꼼꼼히 읽어보면 많은 도움이 될 것 같아요.
- 초반 프로젝트라 기술적인 챌린지에 대해선 넣지 않았는데요. 다음 프로젝트부터는 이 부분도 고려해보시면 좋을 것 같아요. (멘토링 시간에 앞으로의 공부 방향성 및 기술적 고찰에 대해 이야기 해볼게요.)

그동안 프로젝트 하느라 고생 많으셨습니다!

Commit:
fix: 오류 해결완료

Commit:
fix:오류해결

rOyalFruit committed:
Merge branch 'dev' of https://github.com/prgrms-be-devcourse/NBE3-4-2-team8 into Feature1-Book,Cart

rOyalFruit committed:
Merge pull request #44 from prgrms-be-devcourse/feature-order-search

Feature order search

rOyalFruit committed:
fix: 병합 후 생긴 에러 수정

nodonghui committed:
save

nodonghui committed:
merge solve

rOyalFruit committed:
fix: 병합 후 생긴 에러 수정

Commit:
fix:날짜필터 수정

rOyalFruit committed:
Feat: 사이드바에 사용자 프로필 이미지 추가

rOyalFruit committed:
Merge pull request #45 from prgrms-be-devcourse/Feature1-Book,Cart

전반적인 에러 수정, 사이드바에 프로필 이미지 표시 추가

nodonghui committed:
save

nodonghui committed:
save

Commit:
fix:order 페이징처리

Commit:
fix:order 페이징처리

Commit:
fix: order 페이징 처리

Commit:
fix

1m1nkim committed:
Fix: 바로구매 detail_order에 저장 안되던 점 수정

nodonghui committed:
merge solve

BeigeBearDev committed:
feat: admin 부분 병합

BeigeBearDev committed:
Merge branch 'dev' into feature-admin

1m1nkim committed:
Feat: 책 상세페이지 개발, 기본 배송지 주문기능에서 불러오기
Fix: my/orders 사이드바 경로 변경

- 마이페이지 부분 코드 일부 수정,

Commit:
Merge pull request #46 from prgrms-be-devcourse/temp-order

Temp order

BeigeBearDev committed:
Merge pull request #47 from prgrms-be-devcourse/feature-admin

1m1nkim committed:
Merge branch 'dev' into Fe-pr

1m1nkim committed:
Merge pull request #49 from prgrms-be-devcourse/Fe-pr

프론트 작업 pr

nodonghui committed:
fect : 사용자 리뷰 조회 api 구현

nodonghui committed:
merge solve

nodonghui committed:
fect : memberControllerTest 기반 코드 작성

테스트 코드를 실행하려면
application-test.yml을 확인하고 환경 변수 설정을 해야합니다

rOyalFruit committed:
Fix: 관리자 페이지 글자 색이 하얗게 표시되는 문제 수정

rOyalFruit committed:
Merge pull request #48 from prgrms-be-devcourse/Feature1-Book,Cart

Fix: 관리자 페이지 글자 색이 하얗게 표시되는 문제 수정

1m1nkim committed:
"머지 병합 충돌 해결을 위한 파일 생성"

1m1nkim committed:
Merge Conflict Slove

1m1nkim committed:
Merge Conflict Solve 2

nodonghui committed:
test : 사용자, 배송정보, 질문 테스트 코드 추가

MemberController, DeliveryInformationController,
QuestionController 메서드 테스트 코드를 추가했습니다.
예외적인 데이터를 고려한 테스트 코드는 작성하지 않았습니다.
test 실행시 application-test의 환경 변수에 대한 정보를 넣어야 합니다.
2025 - 02 - 17

nodonghui committed:
Merge branch 'dev' of https://github.com/prgrms-be-devcourse/NBE3-4-2-team8 into dev

2025-02-17

nodonghui committed:
fect : 리뷰 조회 기능 구현

마이페이지 리뷰 조회 기능 api와 테스트 코드를 구현했습니다.
프론트 마이페이지 리뷰 내역을 클릭해 확인할수있습니다.
마이페이지 갱신 요청 dto의 imageUrl를 유효성 검사 문제로 임시 주석처리했습니다.
추후 구현할 예정입니다.

nodonghui committed:
fect: 사진 업로드 기능, xss 처리기능 구현

질문 데이터 입력시 content와 title 값에 xss 방지 처리 되어 저장됩니다.
사진 데이터 업로드를 위한 QuestionGenFileController와 유틸 메서드를 만들었습니다.
api 시간 효율 측정을 위한 ExceutionTimeAspect를 추가했습니다
질문 목록 조회 시 사용하는 dto에 answerList를 삭제해 조회 시간을 줄였습니다.

nodonghui committed:
refactor : qustion 리팩토링

sql 최적화 진행

nodonghui committed:
save branch

nodonghui committed:
save

nodonghui committed:
Merge pull request #53 from prgrms-be-devcourse/merge/nodonghui/03-14

Merge/nodonghui/03 14
